core/dbt/parser/unit_tests.py

```diff
@@ -189,6 +190,26 @@ def parse(self) -> ParseResult:
                 unit_test_fqn = [self.project.project_name] + model_name_split + [test.name]
                 unit_test_config = self._build_unit_test_config(unit_test_fqn, test.config)
 
+                # Check that format and type of rows matches for each given input
+                for input in test.given:
+                    if (input.format == "dict" and not isinstance(input.rows, list)) or (
+                        input.format == UnitTestFormat.CSV and not isinstance(input.rows, str)
+                    ):
+                        raise ParsingError(
+                            f"Input rows in invalid format for unit test {test.name}"
+                        )
+                # Check that format type of rows matches for expect
+                if (
+                    test.expect.format == UnitTestFormat.Dict
+                    and not isinstance(test.expect.rows, list)
+                ) or (
+                    test.expect.format == UnitTestFormat.CSV
+                    and not isinstance(test.expect.rows, str)
+                ):
+                    raise ParsingError(
+                        f"Expected rows in invalid format for unit test {test.name}"
+                    )
+
```

Comment:
I think ideally this kind of validation happens on the `UnitTestFixture` itself so callers/users of the instantiated class don't need to worry about it. What do you think about making it a post_init method on UnitTestFixture?

Reply:
I made a separate "validate_fixture" class method and called it when parsing. A post_init method would work too, but I'm just not all that fond of doing validation there. We generally validate as a separate step, so this is a bit more like that.